thirdparty/github.go

```diff
@@ -103,184 +131,200 @@ func GetGithubAPIStatus() (string, error) {
 
 // GetGithubFile returns a struct that contains the contents of files within
 // a repository as Base64 encoded content.
-func GetGithubFile(oauthToken, fileURL string) (githubFile *GithubFile, err error) {
-	resp, err := tryGithubGet(oauthToken, fileURL)
-	if resp == nil {
-		errMsg := fmt.Sprintf("nil response from url '%v'", fileURL)
-		grip.Error(errMsg)
-		return nil, APIResponseError{errMsg}
-	}
-	defer resp.Body.Close()
-
+func GetGithubFile(oauthToken, owner, repo, path, hash string) (*github.RepositoryContent, error) {
+	httpClient, err := util.GetHttpClientForOauth2(oauthToken)
 	if err != nil {
-		errMsg := fmt.Sprintf("error querying '%v': %v", fileURL, err)
-		grip.Error(errMsg)
-		return nil, APIResponseError{errMsg}
+		return nil, errors.Wrap(err, "can't fetch data from github")
 	}
+	defer util.PutHttpClientForOauth2(httpClient)
+	client := github.NewClient(httpClient)
 
-	if resp.StatusCode != http.StatusOK {
-		grip.Errorf("Github API response: ‘%s’", resp.Status)
-		if resp.StatusCode == http.StatusNotFound {
-			return nil, FileNotFoundError{fileURL}
+	var opt *github.RepositoryContentGetOptions
+	if len(hash) != 0 {
+		opt = &github.RepositoryContentGetOptions{
+			Ref: hash,
 		}
-		return nil, errors.Errorf("github API returned status '%v'", resp.Status)
 	}
 
-	respBody, err := ioutil.ReadAll(resp.Body)
+	var file *github.RepositoryContent
+	var resp *github.Response
+	_, _ = util.Retry(
```

Comment:
`_, err` (There are some more later in the file, but I'll stop commenting on them.)